### ISSUE TITLE
Docs: tseslint.config() is still used in generated snippets for extension rules

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#12107]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [x] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Replace with `defineConfig()`

### Affected URL(s)

https://typescript-eslint.io/rules/no-unused-vars/
many others

### Additional Info

Specifically, the "How to Use" section contains:

```ts
export default tseslint.config({
  rules: {
    // Note: you must disable the base rule as it can report incorrect errors
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error"
  }
});
```

Credit to @Azuriru on this discord thread for noticing! https://discord.com/channels/1026804805894672454/1479596810497691750

**Comment by kirkwaiblinger** [comment]
And non-extension rules, heh
